superset/assets/src/SqlLab/components/QueryAutoRefresh.jsx

```diff
@@ -16,12 +17,23 @@ class QueryAutoRefresh extends React.PureComponent {
   componentWillUnmount() {
     this.stopTimer();
   }
+  timeoutOldQueries() {
```

Comment:
This logic should not be in front-end (browser-side). 
Query table knows each query started time. it doesn't need front-end sends ajax call to set query state. 
when front-end ask a query's status, server side should first check the query's start time, if

- longer than 6 hours old: set it timeout and return timeout state.
- < 6 hours: return its current status.